packages/jaeger-ui/src/components/TracePage/CriticalPath/testCases/test1.js

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) 2023 Uber Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import transformTraceData from '../../../../model/transform-trace-data';
+
+/*
+      +──────────────────────+
+      │      span C          │
+      +++──────+++++──────┐+++
+        │span D│   │span E│
+        ++++++++   ++++++++
+
+Here +++++ are critical path sections
+
+*/
+const testTrace = {
+  traceID: 'test1-trace',
+  spans: [
+    {
+      spanID: 'span-E',
+      operationName: 'operation E',
+      references: [
+        {
+          refType: 'CHILD_OF',
+          spanID: 'span-C',
+        },
+      ],
+      startTime: 50,
+      duration: 10,
+      processID: 'p1',
+    },
+    {
+      spanID: 'span-C',
+      operationName: 'operation C',
+      references: [],
+      startTime: 1,
+      duration: 100,
+      processID: 'p1',
+    },
+    {
+      spanID: 'span-D',
+      operationName: 'operation D',
+      references: [
+        {
+          refType: 'CHILD_OF',
+          spanID: 'span-C',
+        },
+      ],
+      startTime: 20,
+      duration: 20,
+      processID: 'p1',
+    },
+  ],
+  processes: {
+    p1: {
+      serviceName: 'customers-service',
+    },
+  },
+};
+
+const transformedTrace = transformTraceData(testTrace);
+
+const sanitizedSpanData = transformedTrace.spans;
+const refinedSpanData = transformedTrace.spans;
+refinedSpanData[0].childSpanIds = ['span-E', 'span-D'];
+refinedSpanData[1].childSpanIds = [];
+refinedSpanData[2].childSpanIds = [];
```

Comment:
there is no need to have this code in the test fixture file, it can be done by a utility function in the test.

Reply:
But my doubt is , This data is required to test the helper functions findRootSpanId  ,  findChildSpanIds etc.. like this. So how can I test them if I don't have these data??

Reply:
You can have separate unit tests that specifically test those functions. But these fixtures are for testing the CP algorithm, not its submodules, so it's not necessary to create these manually, since they are by-product of the algorithm itself.